src/modules/user/user.usecase.ts

```diff
@@ -698,6 +700,72 @@ export class UserUseCases {
     });
   }
 
+  async sendAccountUnblockEmail(email: User['email']): Promise<void> {
+    const secret = this.configService.get('secrets.jwt');
+    const user = await this.userRepository.findByEmail(email);
+
+    if (!user) {
+      throw new NotFoundException();
+    }
+
+    const defaultIat = getTokenDefaultIat();
+
+    await this.userRepository.updateByUuid(user.uuid, {
+      lastPasswordChangedAt: new Date(defaultIat * 1000),
+    });
+
+    const unblockAccountToken = SignWithCustomDuration(
+      {
+        payload: {
+          uuid: user.uuid,
+          email: user.email,
+          action: AccountTokenAction.Unblock,
+        },
+        iat: defaultIat,
+      },
+      secret,
+      '48h',
+    );
+
+    const driveWebUrl = this.configService.get('clients.drive.web');
+    const unblockAccountTemplateId = this.configService.get(
+      'mailer.templates.unblockAccountEmail',
+    );
+
+    const url = `${driveWebUrl}/blocked-account/${unblockAccountToken}`;
+
+    await this.mailerService.send(user.email, unblockAccountTemplateId, {
+      email,
+      unblock_url: url,
+    });
+  }
+
+  async unblockAccount(
+    userUuid: User['uuid'],
+    tokenIat?: number,
+  ): Promise<void> {
+    const user = await this.userRepository.findByUuid(userUuid);
+
+    if (!user) {
+      throw new BadRequestException();
+    }
+
+    if (
+      tokenIat &&
+      !isTokenIatGreaterThanDate(
+        new Date(user?.lastPasswordChangedAt),
+        tokenIat,
+      )
+    ) {
+      throw new ForbiddenException();
+    }
+
+    await this.userRepository.updateByUuid(userUuid, {
+      errorLoginCount: 0,
+      lastPasswordChangedAt: null,
```

Comment:
@sg-gs Do you recommend to set to null the lastPasswordChangedAt field or we can just leave it?

Reply:
We need to expire other sessions for security reasons